tests/nlu/extractors/test_entity_synonyms.py

```diff
@@ -60,7 +60,7 @@ def test_unintentional_synonyms_capitalized(
 
     mapper.train(TrainingData(training_examples=examples))
 
-    assert mapper.synonyms.get("mexican") is None
+    assert mapper.synonyms.get("mexican") == "Mexican"
```

Comment:
If alternate casing can be introduced for synonyms, this test becomes redundant.
Double-checking with @twerkmeister if you foresee an issue with this change in this PR?

Reply:
I don't remember exactly why this test was put in place. What I could imagine is that some acronyms can look like normal words if they are lowercased. But given that the `EntitySynonymMapper` is only applied to something an entity extractor has found before, proper context around the word would have been taken into account. Seems to me it should be okay to be more flexible here.

Reply:
@twerkmeister thanks for the quick reply!

@vcidst We should probably enquire with CSE if they also foresee an issue; if they do then we might have to make this improvement customizable (potentially implement a CLI option in `rasa train` for switching on or off alternate casing?)

Reply:
@ancalita Who should I contact from CSE for feedback? 

> add a boolean option in nlu.yml under synonym to allow for alternate casing
I think it can be done but it will increase the scope of the ticket. Do we need to loop in someone from Product to make a decision or do it based on CSE's feedback?

Reply:
> Who should I contact from CSE for feedback?

You can reach out directly in `cs-engineering` channel, tagging Arjaan.

> Do we need to loop in someone from Product to make a decision or do it based on CSE's feedback?

We should keep Andrea informed and consult with Paul about the design solution, if in the end we're adding customization.

Reply:
Alright, perfect, I only suggested to add a customizable option for switching on/off alternate casing only if CSE thought it was a good idea to have this. Based on the conversation, I see this is not required 👍🏻